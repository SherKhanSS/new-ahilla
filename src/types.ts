export interface Film {
  backgroundColor: string,
  cover: string,
  description: string,
  director: string,
  genre: string,
  id: number,
  isFavorite: boolean,
  poster: string,
  preview: string,
  previewImage: string,
  ratingCount: number,
  ratingScore: number,
  runTime: number,
  starring: string[],
  title: string,
  videoLink: string,
  year: number,
}

export interface Review {
  id: number,
  user: {
    id: number,
    name: string,
  },
  rating: number,
  comment: string,
  date: string,
}
