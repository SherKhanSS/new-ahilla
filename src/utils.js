export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getCurentFilm = (films, props) => {
  return films.find((movie) => movie.id === Number(props.match.params.id));
};

export const noop = () => {
  // do nothing
};
