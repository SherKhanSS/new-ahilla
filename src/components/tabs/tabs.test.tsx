import * as React from "react";
import * as renderer from "react-test-renderer";
import Tabs from "./tabs";
import {Film, Review} from "../../types";
import {noop} from "../../utils";

const reviews: Review[] = [
  {
    id: 1,
    user: {
      id: 4,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];

const film: Film = {
  backgroundColor: `#BDAD8F`,
  cover: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/No_Country_for_Old_Men.jpg`,
  description: `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`,
  director: `Ethan Coen`,
  genre: `Crime`,
  id: 1,
  isFavorite: false,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/No_Country_for_Old_Men.jpg`,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/no-country-for-old-men.jpg`,
  ratingCount: 764976,
  ratingScore: 4.1,
  runTime: 122,
  starring: [`Tommy Lee Jones`, `Javier Bardem`, `Josh Brolin`],
  title: `The Grand Budapest Hotel`,
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  year: 2007,
};

const tabCurrent = `Overview`;

it(`Render Tabs`, () => {
  const tree = renderer
    .create(<Tabs
      tabCurrent={tabCurrent}
      onTabClick={noop}
      film={film}
      reviews={reviews}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
