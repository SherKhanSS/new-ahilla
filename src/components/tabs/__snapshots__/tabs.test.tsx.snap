// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Tabs 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      4.1
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Normal
      </span>
      <span
        className="movie-rating__count"
      >
        764976
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Ethan Coen
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Tommy Lee Jones
        , 
        Javier Bardem
        , 
        Josh Brolin
        , 
      </strong>
    </p>
  </div>
</div>
`;
